impl_core/src/main/java/io/opencensus/implcore/tags/propagation/CorrelationContextFormat.java

```diff
@@ -0,0 +1,175 @@
+  private static final char TAG_KEY_VALUE_DELIMITER = '=';
+  private static final char TAG_DELIMITER = ',';
+  private static final Splitter TAG_KEY_VALUE_DELIMITER_SPLITTER =
+      Splitter.on(TAG_KEY_VALUE_DELIMITER);
+  private static final Splitter TAG_DELIMITER_SPLITTER = Splitter.on(TAG_DELIMITER);
+
+  // private static final char TAG_PROPERTIES_LEFT_PAREN = '[';
+  // private static final char TAG_PROPERTIES_RIGHT_PAREN = ']';
+  // private static final char TAG_PROPERTIES_DELIMITER = ';';
+  // private static final char TAG_PROPERTIES_KEY_VALUE_DELIMITER = '=';
+
+  @VisibleForTesting
+  static final TagMetadata METADATA_UNLIMITED_PROPAGATION =
+      TagMetadata.create(TagTtl.UNLIMITED_PROPAGATION);
+
+  private final CurrentState state;
+
+  CorrelationContextFormat(CurrentState state) {
+    this.state = state;
+  }
+
+  @Override
+  public List<String> fields() {
+    return FIELDS;
+  }
+
+  @Override
+  public <C /*>>> extends @NonNull Object*/> void inject(
+      TagContext tagContext, C carrier, Setter<C> setter) throws TagContextSerializationException {
+    checkNotNull(tagContext, "tagContext");
+    checkNotNull(carrier, "carrier");
+    checkNotNull(setter, "setter");
+    if (State.DISABLED.equals(state.getInternal())) {
+      return;
+    }
+
+    try {
+      StringBuilder stringBuilder = new StringBuilder(TAGCONTEXT_SERIALIZED_SIZE_LIMIT);
+      int totalChars = 0; // Here chars are equivalent to bytes, since we're using ascii chars.
+      int totalTags = 0;
+      for (Iterator<Tag> i = InternalUtils.getTags(tagContext); i.hasNext(); ) {
+        Tag tag = i.next();
+        if (TagTtl.NO_PROPAGATION.equals(tag.getTagMetadata().getTagTtl())) {
+          continue;
+        }
+        if (stringBuilder.length() > 0) {
+          stringBuilder.append(TAG_DELIMITER);
+        }
+        totalTags++;
+        totalChars += encodeTag(tag, stringBuilder);
+      }
+      checkArgument(
+          totalTags <= MAX_NUMBER_OF_TAGS,
+          "Number of tags in the TagContext exceeds limit " + MAX_NUMBER_OF_TAGS);
+      // Note per W3C spec, only the length of tag key and value counts towards the total length.
+      // Length of properties (a.k.a TagMetadata) does not count.
+      checkArgument(
+          totalChars <= TAGCONTEXT_SERIALIZED_SIZE_LIMIT,
+          "Size of TagContext exceeds the maximum serialized size "
+              + TAGCONTEXT_SERIALIZED_SIZE_LIMIT);
+      setter.put(carrier, CORRELATION_CONTEXT, stringBuilder.toString());
+    } catch (IllegalArgumentException e) {
+      throw new TagContextSerializationException("Failed to serialize TagContext", e);
+    }
+  }
+
+  // Encodes the tag to the given string builder, and returns the length of encoded key-value pair.
+  private static int encodeTag(Tag tag, StringBuilder stringBuilder) {
+    String key = tag.getKey().getName();
+    String value = tag.getValue().asString();
+    int charsOfTag = key.length() + value.length();
```

Comment:
Per https://github.com/w3c/correlation-context/blob/master/correlation_context/HTTP_HEADER_FORMAT.md#header-value I don't think so, because it says "name-value pair". /cc @SergeyKanzhelev for confirmation.